server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalDateHistogram.java

```diff
@@ -425,6 +421,14 @@ public void accept(long key) {
 
     private void iterateEmptyBuckets(List<Bucket> list, ListIterator<Bucket> iter, LongConsumer onBucket) {
         LongBounds bounds = emptyBucketInfo.bounds;
+        Rounding.Prepared prepared = null;
+        if (bounds != null) {
+            prepared = createPrepared(bounds.getMin() + offset, bounds.getMax() + offset);
+        } else {
+            if (list.isEmpty() == false) {
+                prepared = createPrepared(list.get(0).key, list.get(list.size() - 1).key);
```

Comment:
Just double checking - if we don't have any explicit bounds set on the agg we'll use the keys of the actual output buckets. Which is fine - because we're not expanding keys. We only do that if there are bounds.

Reply:
>  if we don't have any explicit bounds set on the agg we'll use the keys of the actual output buckets. Which is fine - because we're not expanding keys. We only do that if there are bounds.

Yes, that is how I understand this logic too. Creating of the empty buckets before or after the actual buckets doesn't happen if there are no bounds.